fms/models/hf/utils.py

```diff
@@ -250,31 +211,105 @@ def _infer_model_configuration(
         config_params["use_bias"] = config.mamba_proj_bias
         config_params["norm_eps"] = config.rms_norm_eps
     elif architecture == "SiglipModel":
+        infer_common_params = False
         config = config.vision_config
-        inner_dim = config.intermediate_size
         architecture = "siglip_vision"
         config_params["hidden_size"] = config.hidden_size
         config_params["intermediate_size"] = config.intermediate_size
-        config_params["num_hidden_layers"] = config.num_hidden_layers
-        config_params["num_attention_heads"] = config.num_attention_heads
+        config_params["nlayers"] = config.num_hidden_layers
+        config_params["nheads"] = config.num_attention_heads
         config_params["num_channels"] = config.num_channels
         config_params["image_size"] = config.image_size
         config_params["patch_size"] = config.patch_size
         config_params["hidden_act"] = config.hidden_act
         config_params["layer_norm_eps"] = config.layer_norm_eps
         config_params["attention_dropout"] = config.attention_dropout
+    elif architecture == "LlavaNextForConditionalGeneration":
+        from fms.models.siglip_vision import SiglipVisionConfig
+        from fms.models.granite import GraniteConfig
+
+        if config.text_config.model_type != "granite":
+            raise ValueError(
+                "FMS implementation of LlavaNext currently supports only Granite language model"
+            )
+        if config.vision_config.model_type != "siglip_vision_model":
+            raise ValueError(
+                "FMS implementation of LlavaNext currently supports only Siglip vision model"
+            )
+
+        infer_common_params = False
+        architecture = "llava_next"
+        config_params["image_token_index"] = config.image_token_index
+        config_params["image_grid_pinpoints"] = config.image_grid_pinpoints
+        config_params["vision_feature_layer"] = config.vision_feature_layer
+        config_params["vision_feature_select_strategy"] = (
+            config.vision_feature_select_strategy
+        )
+        _, vision_config_params = _map_model_config("SiglipModel", config)
+        config_params["vision_config"] = SiglipVisionConfig(**vision_config_params)
+        _, text_config_params = _map_model_config(
+            "GraniteForCausalLM", config.text_config
+        )
+        config_params["text_config"] = GraniteConfig(**text_config_params)
+
     else:
         raise ValueError(
-            "FMS model implementations currently only support LlamaForCausalLM, GPTBigCodeForCausalLM, MixtralForCausalLM, RobertaForMaskedLM and GraniteForCausalLM"
+            "FMS model implementations currently only support LlamaForCausalLM, GPTBigCodeForCausalLM, MixtralForCausalLM, RobertaForMaskedLM, GraniteForCausalLM, SiglipModel and LlavaNextForConditionalGeneration"
         )
 
     # infer common params
-    if hasattr(config, "vocab_size"):
+    if infer_common_params:
```

Comment:
even if infer_common_params is True, some models lack src_vocab_size right?

Reply:
That was only true for Siglip, but I took care of it in its section itself.